src/lightning/fabric/fabric.py

```diff
@@ -925,14 +925,22 @@ def _wrap_with_setup(self, to_run: Callable, *args: Any, **kwargs: Any) -> Any:
             return to_run(*args, **kwargs)
 
     def _move_model_to_device(self, model: nn.Module, optimizers: List[Optimizer]) -> nn.Module:
-        initial_device = next(model.parameters(), torch.tensor(0)).device
-        if any(param.device != initial_device for param in model.parameters()):
-            rank_zero_warn(
-                "The model passed to `Fabric.setup()` has parameters on different devices. Since `move_to_device=True`,"
-                " all parameters will be moved to the new device. If this is not desired, set "
-                " `Fabric.setup(..., move_to_device=False)`.",
-                category=PossibleUserWarning,
-            )
+        try:
+            initial_name, initial_param = next(model.named_parameters())
+        except StopIteration:
+            pass
+        else:
+            initial_device = initial_param.device
+            for name, param in model.named_parameters():
+                if param.device != initial_device:
+                    rank_zero_warn(
+                        f"The model passed to `Fabric.setup()` has parameters on different devices ({name!r} on"
+                        f" {param.device} and {initial_name!r} on {initial_device}). Since `move_to_device=True`,"
+                        " all parameters will be moved to the new device. If this is not desired, set "
+                        " `Fabric.setup(..., move_to_device=False)`.",
+                        category=PossibleUserWarning,
+                    )
+                    break
```

Comment:
```suggestion
            wrong_params = {name: param.device for name, param in model.named_parameters() if param.device != initial_device]
            if wrong_params:
                rank_zero_warn(
                    f"The model passed to `Fabric.setup()` has parameters on different devices ({wrong_params!r}"
                    f" and initial parameter on {initial_device}). Since `move_to_device=True`,"
                    " all parameters will be moved to the new device. If this is not desired, set "
                    " `Fabric.setup(..., move_to_device=False)`.",
                    category=PossibleUserWarning,
                )
```

Reply:
Rather say which all are wrong, then push a user to run it many times because you tel him only one at a time...

Reply:
This could be a very very long list depending on the model size. It would be unreadable in those cases.

Reply:
rather have long list once then need to run it 100 times and pay $$

Reply:
This happens at the very beginning of the script, when the model is set up, meaning that it's not something that you'd get hours into a training session.

And normally when this warning happens and it's a true positive, the fix for one will fix it for all others. A simple example of this is a parameter in your transformer block layers. If you have 70 block layers, with your suggestion the same parameter would appear 70 times.

What do you think here @awaelchli?

Reply:
Yes, I'd prefer to print the first detection.

Reply:
so how about say fist and if there are more add a number how many more...

Reply:
I added the total count